pkg/synchromanager/clustersynchro_manager.go

```diff
@@ -234,29 +244,43 @@ func (manager *Manager) reconcileCluster(cluster *clusterv1alpha2.PediaCluster)
 	if synchro != nil && !reflect.DeepEqual(synchro.RESTConfig, config) {
 		klog.InfoS("cluster config is changed, rebuild cluster synchro", "cluster", cluster.Name)
 
-		synchro.Shutdown(false, false)
+		synchro.Shutdown(true, false)
 		synchro = nil
 
+		manager.synchrolock.Lock()
+		manager.synchros[cluster.Name] = synchro
+		manager.synchrolock.Unlock()
+
 		// manager.cleanCluster(cluster.Name)
 	}
 
 	// create resource synchro
 	if synchro == nil {
-		// TODO(iceber): set the stop sign of the manager to cluster synchro
-		synchro, err = clustersynchro.New(cluster.Name, config, manager.storage, manager)
-		if err != nil {
-			// TODO(iceber): update cluster status
-			// There are many reasons why creating a cluster synchro can fail.
-			// How do you gracefully handle different errors?
+		if err = wait.PollImmediate(waitClusterSynchroInternal, waitClusterSynchroTimeout, func() (done bool, err error) {
+			synchro, err = clustersynchro.New(cluster.Name, config, manager.storage, manager)
+			if err == nil {
+				return true, nil
+			}
+
+			manager.UpdateClusterSynchroCondition(cluster.Name, clusterv1alpha2.InitialFailedConditionReason, err.Error(), metav1.ConditionFalse)
 
-			klog.ErrorS(err, "Failed to create cluster synchro", "cluster", cluster.Name)
+			currentCluster, err := manager.clusterlister.Get(cluster.Name)
+			if err != nil || currentCluster.Generation != cluster.Generation {
+				return false, errors.New("cluster config is changed or failed to get cluster")
```

Comment:
If the cluster changes, the cluster name already exists in the queue waiting for the next transfer and does not need to be re-queued

Reply:
The error is not returned for controller requeue. When an error message is received, wait.PollImmediate will stop try and return err,  the err handler will log it and return nill to finish the reconcile progress.